frontend/controller/serviceworkers/sw-primary.js

```diff
@@ -359,6 +366,20 @@ self.addEventListener('message', function (event) {
         }).finally(() => {
           port.close()
         })
+
+        // If the window is outdated (different GI_VERSION), trigger an event
+        // of type 'NOTIFICATION_TYPE.VERSION_INFO'.
+        if (
+          currentVersionInfo &&
+          event.source &&
+          event.data.GI_VERSION !== currentVersionInfo.GI_VERSION
+        ) {
```

Comment:
This logic doesn't make sense to me for a few reasons:
 
- The `currentVersionInfo` received is based on the value set by the `VERSION_INFO` handler above, and yet that value is based on `process.env.GI_VERSION`, which should be available here in the SW too; there's no reason to send it over, we should be able to directly access it there.
- The web socket, which exists in the SW, is what sends over the new version info to the SW. It's not supposed to have anything to do with the browser.

Reply:
> The currentVersionInfo received is based on the value set by the VERSION_INFO handler above, and yet that value is based on process.env.GI_VERSION, which should be available here in the SW too; there's no reason to send it over, we should be able to directly access it there.

The entire point of this logic is to detect version discrepancies. `process.env.GI_VERSION` is indeed available in the SW, but by itself it'd not detect an outdated version.

> The web socket, which exists in the SW, is what sends over the new version info to the SW. It's not supposed to have anything to do with the browser.

It has something to do with the browser in that the browser triggers a refresh.

Reply:
> The entire point of this logic is to detect version discrepancies. process.env.GI_VERSION is indeed available in the SW, but by itself it'd not detect an outdated version

The problem is that if I load the page in a browser, my `process.env.GI_VERSION` may well be the latest one, right? But the SW might be running on the old one. Its `process.env.GI_VERSION` would be out of date.

So this check:

```js
event.data.GI_VERSION !== currentVersionInfo.GI_VERSION
```

Would not get triggered, because they'd be the same value, but the check should be:

```js
currentVersionInfo.GI_VERSION !== process.env.GI_VERSION
```

Reply:
Ok... so this code is really useful only if the page that's loaded is loaded from cache, and therefore has an outdated `process.env.GI_VERSION` that it sends.

OK, could you add that as a comment to explain that?